reana_workflow_controller/consumer.py

```diff
@@ -281,7 +291,7 @@ def _update_job_cache(msg):
     Session.add(cached_job)
 
 
-def _delete_workflow_job(workflow: Workflow) -> None:
+def _delete_workflow_batch_pod(workflow: Workflow) -> None:
```

Comment:
Please beware of the terminology when renaming. We are deleting Kuberentes "job" here in this function, not only "pod", when we are doing `delete_namespaced_job()`. So this rename wouldn't be fully justified. 

When a workflow starts, it creates a Kubernetes job named such as 'reana-run-batch-0bf3cf48-247a-4cbf-bd5f-727cf41e5edb` which will create a Kubernetes pod named such as `reana-run-batch-0bf3cf48-247a-4cbf-bd5f-727cf41e5edb-hz46p` where the workflow engine runs. 

It is the latter that can get to `NotReady` status situations. When this happens, the manual solution is to delete the former (=job), which will then automatically delete also the latter (=pod). (If we would have deleted only the batch pod, then the kubernetes job would start a new batch pod.)

Perhaps the terminology is confusing since we have both Kubernetes job and pod, and workflow engine job and pod, and that workflow engine pod can generate new workflow step jobs and pods... Needs some renaming care....

Reply:
Yeah. It is not a final naming here. `_delete_workflow_batch_job` is a better name or even `_delete_workflow_engine_batch_job`.

And, I would agree terminology is confusing.